src/hooks/core/chat/useMoveChatTypeHook.ts

```diff
@@ -0,0 +1,173 @@
+
+    const isAnonymous = channelCategory === ANONYMOUS;
+    const type: {[key: string]: ChannelType} = {
+      messaging: isAnonymous ? 'ANON_PM' : 'PM',
+      group: 'GROUP',
+      topics: 'TOPIC'
+    };
+
+    let signedChannelName;
+    let signedChannelImage;
+
+    if (!isAnonymous) {
+      const myUserData = channel?.members?.find(
+        (member) => member?.user?.id === signedProfileId
+      )?.user;
+      signedChannelName =
+        channel?.channel_type === 4
+          ? `Anonymous ${channel?.anon_user_info_emoji_name}`
+          : getChatName(channel?.name, myUserData?.username ?? myUserData?.name);
+
+      signedChannelImage =
+        channel?.type === 'group' || channel?.type === 'topics'
+          ? channel?.channel_image
+          : channel?.members?.find((member) => member?.user_id !== signedProfileId)?.user?.image ??
+            DEFAULT_PROFILE_PIC_PATH;
+
+      channel.myUserId = signedProfileId;
+    } else {
+      channel.myUserId = undefined;
+    }
+
+    const chatName = isAnonymous
+      ? await getAnonymousChatName(channel?.members)
+      : {name: signedChannelName, image: signedChannelImage};
+
+    channel.targetName = chatName?.name;
+    channel.targetImage = chatName?.image;
+    channel.firstMessage = isAnonymous
+      ? channel?.messages?.[0]
+      : channel?.messages?.[channel?.messages?.length - 1];
+    channel.channel = {...channel};
+
+    const channelType = channel?.type;
+    const channelList = ChannelList.fromChannelAPI(channel, type[channelType], channel?.members);
+
+    await channelList.saveIfLatest(localDb);
+    refresh('channelList', 'chat', 'channelInfo', 'channelMember');
+    goToMoveChat(channelList);
+
+    return channelList;
+  };
+
+  const saveChannelData = async (channel: any, channelCategory: ChannelCategory): Promise<void> => {
+    if (!channel?.members?.length) {
+      console.info('no members');
+      return;
+    }
+
+    if (channel?.type === 'topics') {
+      console.info('type is topics');
+      return;
+    }
+
+    try {
+      await Promise.all(
+        (channel?.members || []).map(async (member) => {
+          const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+          const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
+            channel?.id,
+            uuid(),
+            member
+          );
+          await userMember.saveOrUpdateIfExists(localDb);
+          await memberSchema.save(localDb);
+        })
+      );
+
+      await Promise.all(
+        (channel?.messages || []).map(async (message) => {
+          if (message?.type === 'deleted') return;
+          const chat = ChatSchema.fromGetAllChannelAPI(channel?.id, message);
+          await chat.saveIfNotExist(localDb);
+        })
+      );
+
+      await saveChannelList(channel, channelCategory);
+    } catch (e) {
+      console.log('error on saveChannelData', e);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the error logging could be more descriptive. Also, consider using `console.error` instead of `console.log` for error messages.
```diff
-      console.log('error on saveChannelData', e);
+      console.error('Error occurred while saving channel data:', e);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->